api/views/share.py

```diff
@@ -19,13 +22,35 @@
 class ShareObjectViewSet(viewsets.ReadOnlyModelViewSet):
     ordering = ('-id', )
     pagination_class = CursorPagination
+    permission_classes = [IsDeletedPremissions, ]
```

Comment:
nitpick: Rather than using lists here, I prefer to use tuples. `(item1, )` or `(item1, item2, ...)`

Lists are mutable objects, someone could accidentally call `ShareObjectViewSet.permission_classes.append(...)` anywhere in the code, leading to unexpected errors.

If you're attribute should not be mutable, like `permission_classes` or `ordering`,  tuples give some peace of mind and indicate that the attribute should not be changed.